### PR TITLE
(FM-8422) Make library releasable as a Puppet module

### PR DESCRIPTION
**Pull request by michaeltlombardi**
Prior to this commit the library code could only be released
as a ruby gem. This commit adds the necessary metadata and
documentation for it to be released as a Puppet module upon
which other Puppet modules may depend and thereby use the
manager and helpers.

### CODE REVIEW

**Comment by michaeltlombardi** [comment]
Open question: Do we actually want to do a build + release of a the puppet module including all the cruft here, or do we want a rake task that can clean things up for a much slimmer release (which includes only the library, changelog, appropriate readme, etc, dropping the spec/images/etc).

File: metadata.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
+{
+  "name": "puppetlabs-pwshlib",
+  "version": "0.1.0",
+  "author": "puppetlabs",
+  "summary": "Provide library code for interoperting with PowerShell.",
```
Comment (DavidS) [inline metadata.json]: "interoperating"

File: pwshlib.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# pwshlib
+
+This module enables you to leverage the `ruby-pwsh` gem to execute PowerShell from within your Puppet providers without having to instantiate and tear down a PowerShell process for each command called.
```
Comment (DavidS) [inline pwshlib.md]: The forge will still show the regular README.md

**Comment by DavidS** [comment]
If you want to slim down the module tarball, have a look at the `.pmtignore` which can be used to flag files for ignoring when building into the module.

Either way, as a 0.1.0, this is good enough for me.

File: pwshlib.md (hunk 2 above)
Comment (michaeltlombardi) [reply pwshlib.md]: Yeah, I had a think about this last night, this is one of the problems with having it all in one repo.

I think a build task could rejigger things long enough to do a pdk build, then put them back, but I don't think rewriting the main readme to make more sense on the forge is the right move.

**Comment by michaeltlombardi** [comment]
Added a `.pmtignore` and cleaned up the metadata whoopsy.

File: .pmtignore [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+.rspec_status
+.rubocop.yml
+.travis.yml
+CODEOWNERS
+design-comms.png
+DESIGN.md
+Gemfile
+Gemfile.lock
+Rakefile
+ruby-pwsh.gemspec
```
Comment (RandomNoun7) [inline .pmtignore]: Nit.

**Pull request merged**